src/System.CommandLine.Subsystems/Subsystems/Annotations/AnnotationStorageExtensions.cs

```diff
@@ -0,0 +1,142 @@
+    {
+        symbol.SetAnnotation(annotationId, value);
+        return symbol;
+    }
+
+    /// <summary>
+    /// Attempts to get the value for the annotation <paramref name="id"/> associated with the <paramref name="symbol"/>,
+    /// first from the optional <paramref name="provider"/>, and falling back to the internal annotation storage used to
+    /// store values set via <see cref="SetAnnotation{TValue}(CliSymbol, AnnotationId{TValue}, TValue)"/>.
+    /// </summary>
+    /// <typeparam name="TValue">The type of the annotation value</typeparam>
+    /// <param name="symbol">The symbol that is annotated</param>
+    /// <param name="id">
+    /// The identifier for the annotation. For example, the annotation identifier for the help description is <see cref="HelpAnnotations.Description">.
+    /// </param>
+    /// <param name="value">The annotation value, if successful, otherwise <c>default</c></param>
+    /// <param name="provider">
+    /// An optional annotation provider that may implement custom or lazy construction of annotation values. Annotation returned by an annotation
+    /// provider take precedence over those stored in internal annotation storage.
+    /// </param>
+    /// <returns>True if successful</returns>
+    public static bool TryGetAnnotation<TValue>(this CliSymbol symbol, AnnotationId<TValue> annotationId, [NotNullWhen(true)] out TValue? value, IAnnotationProvider? provider = null)
+    {
+        if (provider is not null && provider.TryGet(symbol, annotationId, out value))
+        {
+            return true;
+        }
+
+        if (symbolToAnnotationStorage.TryGetValue(symbol, out var storage) && storage.TryGet (symbol, annotationId, out value))
+        {
+            return true;
+        }
+
+        value = default;
+        return false;
+    }
+    /// <summary>
+    /// Attempt to retrieve the <paramref name="symbol"/>'s value for the annotation <paramref name="id"/>
+    /// from the optional <paramref name="provider"/> and the internal annotation storage.
+    /// </summary>
+    /// <typeparam name="TValue">The type of the annotation value</typeparam>
+    /// <param name="symbol">The symbol that is annotated</param>
+    /// <param name="id">
+    /// The identifier for the annotation. For example, the annotation identifier for the help description is <see cref="HelpAnnotations.Description">.
+    /// </param>
+    /// <param name="provider">
+    /// An optional annotation provider that may implement custom or lazy construction of annotation values. Annotation returned by an annotation
+    /// provider take precedence over those stored in internal annotation storage.
+    /// </param>
+    /// <returns>The annotation value, if successful, otherwise <c>default</c></returns>
+    public static TValue? GetAnnotationOrDefault<TValue>(this CliSymbol symbol, AnnotationId<TValue> annotationId, IAnnotationProvider? provider = null)
```

Comment:
I am a bit worried about this name as `default` has additional meaning in System.CommandLine

Reply:
Yes, but I'm not sure of a better name. This naming pattern is used by LINQ for `SingleorDefault` etc.